src/clj/game/cards/programs.clj

```diff
@@ -236,19 +236,21 @@
                  :choices {:req #(and (is-type? % "Program")
                                       (installed? %))}
                  :msg (msg "host " (:title target) (when (-> target :cost pos?) ", lowering its cost by 1 [Credit]"))
-                 :effect (effect (host card target)
-                                 (when (-> target :cost pos?)
-                                   (gain state side :credit 1))
-                                 (gain :memory (:memoryunits target))
-                                 (update! (assoc (get-card state card) :dheg-prog (:cid target))))}
+                 :effect (req (gain state side :memory (:memoryunits target))
+                              (when (-> target :cost pos?)
```

Comment:
The trick is that I keep needing to get an updated card state. Not crazy about how this is done, but didn't find the inspiration to greatly change it. Otherwise it would just end up in a `let`.

Reply:
Sorry, you meant the single line thread. I'm ambivalent on that one. I thought you meant my terrible nested update function.